src/sdk/namespace_blob.js

```diff
@@ -459,15 +458,15 @@ class NamespaceBlob {
         const md5_hash = crypto.createHash('md5');
         const block_list = _.times(num_parts, num => {
             const part = part_by_num[num + 1];
-            md5_hash.update(part.etag);
```

Comment:
You've remoted the update of the md5_hash.
On complete_object_upload you return an empty md5_hash.
`
res.etag = md5_hash.digest('hex') + '-' + num_parts;
return res;
`